--- a/src/common/platform.ads
+++ b/src/common/platform.ads
@@ -30,7 +30,8 @@
 
    type Host_Operating_System_Flavor is
       (X86_Windows, X86_64_Windows, X86_Linux, X86_64_Linux, X86_64_Darwin,
-       X86_64_FreeBSD, CodePeer_OS, AArch64_Darwin, AArch64_Linux);
+       X86_64_FreeBSD, CodePeer_OS, AArch64_Darwin, AArch64_Linux,
+       PPC64le_Linux, S390x_Linux);
 
    function Get_OS_Flavor return Host_Operating_System_Flavor;
 
--- /dev/null
+++ b/src/common/ppc64le-linux/platform.adb
@@ -0,0 +1,12 @@
+package body Platform is
+
+   -------------------
+   -- Get_OS_Flavor --
+   -------------------
+
+   function Get_OS_Flavor return Host_Operating_System_Flavor is
+   begin
+      return PPC64le_Linux;
+   end Get_OS_Flavor;
+
+end Platform;
--- /dev/null
+++ b/src/common/s390x-linux/platform.adb
@@ -0,0 +1,12 @@
+package body Platform is
+
+   -------------------
+   -- Get_OS_Flavor --
+   -------------------
+
+   function Get_OS_Flavor return Host_Operating_System_Flavor is
+   begin
+      return S390x_Linux;
+   end Get_OS_Flavor;
+
+end Platform;
--- a/src/gnatprove/spark_report.adb
+++ b/src/gnatprove/spark_report.adb
@@ -1107,7 +1107,9 @@
                 | X86_64_Windows => "Windows",
              when X86_Linux
                 | X86_64_Linux
-                | AArch64_Linux  => "Linux",
+                | AArch64_Linux
+                | PPC64le_Linux
+                | S390x_Linux    => "Linux",
              when X86_64_Darwin  => "Darwin",
              when X86_64_FreeBSD => "FreeBSD",
              when CodePeer_OS    => "CodePeer OS",
