--- a/testsuite/gnatprove/lib/python/test_support.py
+++ b/testsuite/gnatprove/lib/python/test_support.py
@@ -18,7 +18,7 @@
 default_vc_timeout = 120
 parallel_procs = 1
 default_project = "test.gpr"
-default_provers = ["cvc5", "altergo", "z3", "colibri"]
+default_provers = ["cvc5", "altergo", "z3"]
 provers_output_regex = re.compile(
     r"\((Trivial|Interval|CVC4|CVC5|Z3|altergo|colibri).*\)"
 )
