--- a/Makefile
+++ b/Makefile
@@ -107,12 +107,6 @@
 	$(CP) share/spark/theories/*why $(THEORIESDIR)
 	$(CP) share/spark/theories/*mlw $(THEORIESDIR)
 	$(CP) share/spark/runtimes/README $(RUNTIMESDIR)
-	@echo "Generate Coq files by preprocessing context files:"
-	$(MAKE) -C include generate
-	$(CP) include/src/*.ad? $(INCLUDEDIR)
-	$(CP) include/*.gpr $(LIBDIR)
-	$(CP) include/*.gpr.templ $(LIBDIR)
-	$(CP) include/proof $(LIBDIR)
 
 doc: $(DOC)
 
