--- a/src/gnatprove/configuration.adb
+++ b/src/gnatprove/configuration.adb
@@ -1267,7 +1267,7 @@
       if Alt_Ergo /= null then
          Ada.Text_IO.Put (Alt_Ergo.all & ": ");
          Call_With_Status (Alt_Ergo.all,
-                           Arguments => Dash_Dash_Version,
+                           Arguments => Dash_Version,
                            Status    => Status);
          Free (Alt_Ergo);
       end if;
