--- a/why3/src/server/cpulimit-unix.c
+++ b/why3/src/server/cpulimit-unix.c
@@ -11,6 +11,8 @@
 
 #ifndef _WIN32
 
+#define UNUSED(x) (void)(x)
+
 #include <sys/types.h>
 #include <sys/time.h>
 #include <time.h>
@@ -42,7 +44,8 @@
   }
 }
 
-void wallclock_timelimit_reached() {
+void wallclock_timelimit_reached(int signum) {
+  UNUSED(signum);
   fprintf(stderr,
           "Why3cpulimit: wallclock timelimit %d reached, killing command\n",
           wallclock_timelimit);
